lib/Parse/ParseExpr.cpp

```diff
@@ -2905,9 +2910,67 @@ ExprResult Parser::ParseBoundsExpression() {
     SkipUntil(tok::r_paren, StopAtSemi | StopBeforeMatch);
 
   PT.consumeClose();
+
+  if (StartsRelativeBoundsClause(Tok) && !Result.isInvalid()) {
```

Comment:
I think invalidate check of Result should be if(StartRelativeBoundsClause(Tok)). Cause the input parameter of ParseRelativeBoundsClause() is Result.get(). So... I will change this as below.

  if (StartsRelativeBoundsClause(Tok) ) {    // check next token is relative bounds clause.
       if(Result.isInvalid()) { // skip relative bounds clause even if it is invalid result.
           SkipUntil(tok::r_paren, StopAtSemi | StopBeforeMatch);  
       } else { //if it is valid, then
           if( Range Bounds Expression) { // check if it is range bounds expression.
              if( ParseRelativeBoundsClause(Result.get())) // if it is range bounds expression do parsing relative bounds clause
                   Diag(error);
            }else { //if it is not the range bounds expression 
               SkipUntile( r_paren); // skip relative bounds clause
            }
        }